hibernate-core/src/main/java/org/hibernate/query/sqm/tree/select/SqmDynamicInstantiation.java

```diff
@@ -141,7 +140,23 @@ public boolean checkInstantiation(TypeConfiguration typeConfiguration) {
 		}
 	}
 
-	private List<Class<?>> argumentTypes() {
+	public List<MismatchConstructorFieldPositionInfo> findMismatchConstructorFieldPositions(TypeConfiguration typeConfiguration) {
+		if (getInstantiationTarget().getNature() == CLASS) {
+			if (getJavaType().isArray()) {
+				// hack to accommodate the needs of jpamodelgen
+				// where Class objects not available during build
+				return null;
+			}
+			final List<Class<?>> argTypes = argumentTypes();
+
+			return findMismatchConstructorFieldPosition(getJavaType(), argTypes, typeConfiguration);
+		} else {
```

Comment:
This will be causing a checkstyle  failure.
```suggestion
		}
		else {
```